shared/src/iosMain/kotlin/com/twilio/passkeys/AuthorizationControllerWrapper.kt

```diff
@@ -0,0 +1,173 @@
+package com.twilio.passkeys
+
+import com.twilio.passkeys.exception.TwilioException
+import com.twilio.passkeys.exception.UNKNOWN_ERROR
+import com.twilio.passkeys.models.AuthenticatePasskeyResponse
+import com.twilio.passkeys.models.CreatePasskeyResponse
+import com.twilio.passkeys.utils.DeviceUtils
+import platform.AuthenticationServices.ASAuthorization
+import platform.AuthenticationServices.ASAuthorizationController
+import platform.AuthenticationServices.ASAuthorizationControllerDelegateProtocol
+import platform.AuthenticationServices.ASAuthorizationPlatformPublicKeyCredentialAssertion
+import platform.AuthenticationServices.ASAuthorizationPlatformPublicKeyCredentialRegistration
+import platform.AuthenticationServices.ASAuthorizationPublicKeyCredentialAttachment
+import platform.Foundation.NSError
+import platform.darwin.NSObject
+
+interface IAuthorizationControllerWrapper {
+  fun createPasskey(
+    authController: ASAuthorizationController,
+    completion: (CreatePasskeyResult) -> Unit,
+  )
+
+  fun authenticatePasskey(
+    authController: ASAuthorizationController,
+    completion: (AuthenticatePasskeyResult) -> Unit,
+  )
+}
+
+/**
+ * Wraps the ASAuthorizationController functionality in order to improve testability & maintainability.
+ *
+ * @property deviceUtils The utility class for device-related operations.
+ */
+class AuthorizationControllerWrapper : IAuthorizationControllerWrapper {
+  private var authController: ASAuthorizationController? = null
```

Comment:
Could this be a lateinit var?

Reply:
Done, seems way clear with that one. Thank you